hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -163,6 +164,45 @@ object TableNativeReader {
   }
 }
 
+case class PartitionNativeReader(spec: AbstractTypedCodecSpec, fullRowType: TStruct) extends PartitionReader {
```

Comment:
Small note: is there a reason the native partition reader explicitly takes the full row type in the constructor? It should be defined as `encodedVirtualType` in the TypedCodecSpec.

Reply:
this is the requested type probably

Reply:
As I read it, `fullRowType` is the type of the encoded data, and the requested type is passed in as arguments to various methods when needed and isn't stored with the reader.

Reply:
oh, I wasn't reading closely enough. I think you're right.

Reply:
Arcturus was right, I can get the full type from the spec.  Fixed. Thanks!